### ISSUE TITLE
Issue with ;shape and short leather cords

### ISSUE THREAD

**Issue by jonathan-stricker** [rpherbig/dr-scripts#4324]
I had to step away a moment and Fistoskarr was helping me with this on lnet, but there's an issue with how shaping handles a recipe that calls for a "short leather cord", namely the wooden circlet. The script buys the cord correctly, but when it's time to install it, it tries to get a "leather cord" when the correct adjective is a "short cord".

**Comment by jandersson** [comment]
Issue has appeared before. Can you post a log to help the troubleshooting?

**Comment by jonathan-stricker** [comment]
Shoot sorry my log scrolled away, let me run it again and will generate a new log

**Comment by jonathan-stricker** [comment]
The issue happens when you go to get the cord during recipe creation; it tries to "get leather cord" instead of a short cord; it doesn't find the leather cord, thinks you don't have the correct cord, and it aborts. it then tries to go ahead with the workorder, bundling the product into the logbook, which fails, and then it goes onto the next product but hands are full and it stalls out. Will post a log shortly

**Comment by jonathan-stricker** [comment]
Got it, see attached log

[shaping_error.txt](https://github.com/rpherbig/dr-scripts/files/4712906/shaping_error.txt)

**Comment by jandersson** [comment]
So you have some strange typeahead going on thats not related to room movement. It is possibly breaking things. If you know whats causing that, I'd suggest removing it to see if the problem persists.

```
[shape]>study my book

You scan the circlet instructions with a glance and completely understand all facets of the design.
You now feel ready to begin the crafting process.
Roundtime: 8 sec.
You feel fully rested.
R>read my book
...wait 2 seconds.

[shape]>put my book in my haversack

>read my book
>read my book
>read my book
...wait 1 seconds.
...wait 1 seconds.
...wait 1 seconds.
...wait 1 seconds.

[shape]>put my book in my haversack

>read my book

[shape]>put my book in my haversack

[shape]>put my book in my haversack

[shape]>put my book in my haversack

You put your book in your purple haversack.
But you're not holding it.
What were you referring to?
What were you referring to?
What were you referring to?
>read my book

[shape]>get my drawknife

But you're not holding it.
You get a sharpened steel drawknife from inside your purple haversack.

[shape]>get my maple lumber

You need a free hand to pick that up.
You feel fully prepared to cast your spell.

[shape: *** No match was found after 15 seconds, dumping info]

[shape: messages seen length: 16]

[shape: message: You feel fully prepared to cast your spell.]

[shape: message: Sanctify Pattern  (27 roisaen)]

[shape: message: Centering  (OM)]

[shape: message: Minor Physical Protection  (OM)]

[shape: message: Divine Radiance  (OM)]

[shape: message: Benediction  (OM)]

[shape: message: Major Physical Protection  (OM)]

[shape: message: Bless  (OM)]

[shape: message: Gauge Flow  (65 roisaen)]

[shape: message: Protection from Evil  (OM)]

[shape: message: Persistence of Mana  (OM)]

[shape: message: Osrel Meraud  (74%)]

[shape: message: Manifest Force  (26 roisaen)]

[shape: message: [DRPrime]-DR:Samantaka: "Mit, you were here in Oklahoma and didn't even say hi?  :("]

[shape: message: You need a free hand to pick that up.]

[shape: checked against [/^You get/i, /^You are already/i, /^What do you/i, /^What were you/i, /You pick up/i, /can't quite lift it/i]]

[shape: for command get my maple lumber]

[shape]>cast
```

**Comment by jonathan-stricker** [comment]
yeah I noticed there were some type-ahead issues, I'm not sure what is causing it; it doesn't seem to be doing it with my other characters and it doesn't do it to this character all the time, so I'm not sure what's up. is there a yaml setting that impacts type ahead or something?

**Comment by jandersson** [comment]
No idea, that sounds really strange

My suggestion: 
1. Make a copy of your current lich.rbw as lich.rbw.bak (or whatever you like for a name)
2. set `;e use_lich_main`
3. If that doesn't work grab lich from lichproject.org and swap in the lichproject.org version of lich.rbw in there

**Comment by jandersson** [comment]
Just a disclaimer: you should definitely be using `;e use_lich_fork`  in a normal case.

Since I suggested you switch to main for testing, I want to remind that you should switch back to the fork when done testing! 

/disclaimer

**Comment by jonathan-stricker** [comment]
thanks, I'll try to do some of this troubleshooting and report back. haha so noted, thanks! other than just calling ;e use_lich_fork and ;e use_main_fork, is there any way to just check and see what fork I'm on?

**Comment by jandersson** [comment]
`;e echo LICH_VERSION`

If you are on the fork it will have an `f` at the end of it.

On main:
```
>;e echo LICH_VERSION        
--- Lich: exec1 active.      
[exec1: 4.6.52]              
--- Lich: exec1 has exited.  
```

Fork!
```
>;e echo LICH_VERSION        
--- Lich: exec1 active.      
[exec1: 4.12.1f]              
--- Lich: exec1 has exited.  
```

**Comment by jonathan-stricker** [comment]
got it, thanks. yeah I'm on the fork right now, I'll run your troubleshooting suggestions and see what happens, will report back when I get a chance!

**Comment by jonathan-stricker** [comment]
I just switched to use_lich_main and tried to run ;workorder shaping again, same result. The type-ahead seemed more or less the same, and the script still tries to "get leather cord" when it's not a leather cord, it's a short cord. So this line of troubleshooting seems like a dead-end

**Comment by jonathan-stricker** [comment]
I'm swapping out my lich.rbw file for the one from lichproject.org, will report back after I do a little testing

**Comment by jonathan-stricker** [comment]
I'm running some scripts here and there and it still seems pretty aggressive with the type ahead. I noticed when I start up I get this error message:

--- Lich: could not find script 'list' in directory C:/lich/scripts

Is that normal? It's probably unrelated. I'll keep investigating but I'm a little stumped. I'll keep an eye out for issues and just work around them. Easy enough to just comment out the circlet on my shaping list.

**Comment by jandersson** [comment]
>  So this line of troubleshooting seems like a dead-end

Agreed, sorry about derailing this. It's likely the data file causing woes.

**Comment by jandersson** [comment]
The data file looks correct - a wooden circlet specifies a short cord, which you buy:
```
[workorders]>buy short cord

You decide to purchase the cord, and pay the sales clerk 150 Kronars.
The sales clerk hands you your short cord.

[workorders]>put my short cord in my haversack

You put your cord in your purple haversack.
```

but for some reason trying to get a leather cord here
```
[shape]>shape my circlet with my shaper

With short strokes you shape an unfinished maple circlet with your shaper.  Each motion adds more graceful style and exceptional accuracy.

The wood is ready to have more fine detail carved with a carving knife.
Roundtime: 18 sec.

You need another finished short leather cord to continue crafting an unfinished maple circlet.  You believe you can assemble the two ingredients together once you acquire them.

[Ingredients can be added by using ASSEMBLE Ingredient1 WITH Ingredient2]
You feel fully rested.

[shape]>put my beveled shaper in my haversack

You put your shaper in your purple haversack.

[shape]>get my carving knife

You get a steel carving knife from inside your purple haversack.

[shape]>put my carving knife in my haversack

You put your knife in your purple haversack.

[shape]>get my leather cord

What were you referring to?
You feel fully prepared to cast your spell.

[shape: ]
```

**Comment by jandersson** [comment]
Wonder if its this match `You need another finished short leather cord to continue crafting an unfinished maple circlet` causing the issue

**Comment by jonathan-stricker** [comment]
yeah exactly, that's what I noticed. DR does have a tendency to tweak descriptions here and there a little bit. maybe the match list from ;shape needs to be updated to account for the output from that final shape step?

**Comment by jandersson** [comment]
Youre right, the match list needs tweaking (https://github.com/rpherbig/dr-scripts/blob/master/shape.lic#L56)

 The regular expression used right now gets 'leather cord'
https://rubular.com/r/zYQG67pWL03D3L

OK, found the problem, now how to fix without breaking everything 🤔

**Comment by jonathan-stricker** [comment]
Right! Well I'm not that good with regular expressions yet so I leave it in your hands, though if you want to send me a script I can test it out before you push to dependency. :)

**Comment by jonathan-stricker** [comment]
Right on, glad to test it out. will stand by for your ping, thanks so much!

**Comment by MahtraDR** [comment]
Closing as resolved by #5846
